### DISCUSSION TITLE
Declare imported package as not having side effects

### DISCUSSION THREAD

**Discussion by TheUnlocked** [webpack/webpack#13300] [Q&A] [answered]
I have two files set up similarly to this (though this is obviously simplified):
```js
// util.ts
import { someFunction } from 'someone-elses-package';

export function foo() {
    return someFunction();
}

export function bar() {
    return "Bar!";
}

// index.ts (entry point)
import { bar } from './util';

console.log(bar());
```

When I run webpack, it properly eliminates the definition of `foo`, but `someone-elses-package` still gets bundled. I can go into `node_modules` and manually add `"sideEffects": false` to its `package.json` so that the unused package gets eliminated, but that's not very robust way to handle things. Is there some mechanism that allows me to tell webpack to treat an external module as side-effect-free without modifying the package itself?

**Comment by alexander-akait** [comment]
Using `webpackExports` https://webpack.js.org/api/module-methods/#magic-comments

**Comment by sokra** [comment, marked as answer]
You can wrap someoneelses-package in a local module and re-export all exports

**Reply by TheUnlocked** [reply]
Thank you, this worked perfectly!

Note to future readers: remember to label your own package's side effects when doing this. I got confused for a few minutes as to why this suggestion wasn't working because I had forgotten to. 😅
